### PR TITLE
Use preact-modal in adminModal 

### LINKED ISSUE #16328: Convert modals that currently aren't using the Preact modal to the Preact modal
## Description

There's work being done in #16310 for modals and one thing that came up was that there are still some modals in the codebase that aren't using the Preact modal component.

These non-Preact modals should be converted to the Preact modal component, not just for keeping things uniform, but also because the Preact modal component has a tonne of accessibility baked in to it.

_Originally posted by @nickytonline in https://github.com/forem/forem/pull/16310#r793649624_

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR replaces regular modal with preact-modal in adminModal file.

## Related Tickets & Documents

- Related Issue #16328
- Closes #16328

## QA Instructions, Screenshots, Recordings

**How to test this?**
1. Sign in as admin and visit http://localhost:3000/admin/customization/config
2. Click on `Authentication`
3. Click `Enable`, do not enter keys.
4. Click `Update Settings`
5. A modal will appear with warning

All three-instances where modal is getting used.

Screenshot 2022-10-25 at 2 17 58 PM

Screenshot 2022-10-26 at 2 26 25 PM

Screenshot 2022-10-26 at 2 26 41 PM

### UI accessibility concerns?

None

## Added/updated tests?

- [x] Yes
- [ ] No
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

File: app/javascript/admin/__tests__/controllers/config_controller.test.js [hunk 1]
```diff
@@ -1,34 +0,0 @@
-import { Application } from '@hotwired/stimulus';
-import ConfigController from '../../controllers/config_controller';
-
-describe('ConfigController', () => {
```
Comment (rt4914) [inline app/javascript/admin/__tests__/controllers/config_controller.test.js:4]: Removed this test file as we are using preact-modal now and similar to other instances I have added cypress tests for checking this entire functionality.

**Review by fdocr — CHANGES_REQUESTED** [review]
Thanks for this @rt4914, this modal looks great and it's awesome to see we're closing these loops (refactors) 🙌🏼 

I did find it impossible to confirm the modal actions though (enable/disable auth providers). Here's a screen recording of what I see. I did notice the Cypress tests didn't include a flow that ensures confirming the modal actually take the action requested. Perhaps I'm missing something?

https://user-images.githubusercontent.com/6045239/197902746-e32abee8-1265-48fb-b7be-939584293224.mov

**Comment by rt4914** [comment]
> Thanks for this @rt4914, this modal looks great and it's awesome to see we're closing these loops (refactors) 🙌🏼
> 
> I did find it impossible to confirm the modal actions though (enable/disable auth providers). Here's a screen recording of what I see. I did notice the Cypress tests didn't include a flow that ensures confirming the modal actually take the action requested. Perhaps I'm missing something?
> 
>  confirmation_modal_screen_recording.mov

@fdocr Yes you are right that this feature was missing. I have made following changes:
1. fixed the bug
2. added screenshot of dialogs getting used al all-three places in code
3. added tests to make sure the disable-dialog is also covered.

Thanks.

**Review by fdocr — APPROVED** [review]
Tested and works great. Thanks @rt4914!

**Review by Ridhwana — APPROVED** [review]

**Pull request merged**